### PR TITLE
Garbage collecting images in the Kubelet.

### LINKED ISSUE #157: GC old dead containers and unused images
As far as I can tell, we never call DELETE/RemoveContainer nor RemoveImage. 

We should keep containers around for some amount of time for inspection, copying files out, etc., but should eventually clean them up. 

We could treat images as a cache, evicting on LRU when we reach some predetermined space limit -- ideally before the disk just fills up.

### PR DESCRIPTION
**Pull request by vmarmol**
Integrated the imageManager into the Kubelet and applies the garbage
collection policy every 5 minutes. The default policy allows up to 90%
disk usage, after which images are garbage collected to bring limit back
down to 80%.

Fixes #157.

### CODE REVIEW

**Comment by vmarmol** [comment]
e2e is all green :)

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -133,6 +136,8 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Var(&s.ClusterDNS, "cluster_dns", "IP address for a cluster DNS server.  If set, kubelet will configure all containers to use this for DNS resolution in addition to the host's DNS servers")
 	fs.BoolVar(&s.ReallyCrashForTesting, "really_crash_for_testing", s.ReallyCrashForTesting, "If true, crash with panics more often.")
 	fs.DurationVar(&s.StreamingConnectionIdleTimeout, "streaming_connection_idle_timeout", 0, "Maximum time a streaming connection can be idle before the connection is automatically closed.  Example: '5m'")
+	fs.IntVar(&s.ImageGCHighThresholdPercent, "image_gc_high_threshold", s.ImageGCHighThresholdPercent, "The percent of usage after which image garbage collection is always run. Default: 90%%")
```
Comment (jlowdermilk) [inline cmd/kubelet/app/server.go]: Nit: s/usage/disk usage/ for clarity.

**Comment by jlowdermilk** [comment]
lgtm, but @dchen1107 or @saad-ali should probably look it over.

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -88,16 +90,17 @@ func NewKubeletServer() *KubeletServer {
 		EnableServer:          true,
 		Address:               util.IP(net.ParseIP("127.0.0.1")),
 		Port:                  ports.KubeletPort,
-		PodInfraContainerImage:  kubelet.PodInfraContainerImage,
-		RootDirectory:           defaultRootDir,
-		RegistryBurst:           10,
-		EnableDebuggingHandlers: true,
-		MinimumGCAge:            1 * time.Minute,
-		MaxPerPodContainerCount: 5,
```
Comment (saad-ali) [inline cmd/kubelet/app/server.go]: Was `MaxPerPodContainerCount` removed on purpose?

Comment (vmarmol) [inline cmd/kubelet/app/server.go]: Good catch! It was not, that was an incorrect merge. Re-added.

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (vmarmol) [inline cmd/kubelet/app/server.go]: Done.

**Comment by saad-ali** [comment]
LGTM

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**